### PR TITLE
test - siphash-internal

### PR DESCRIPTION
**Pull request by pauli-oracle**
- [x] tests are added or updated

Update the internal siphash tests to use the framework's output..

### CODE REVIEW

File: test/siphash_internal_test.c [hunk 1]
```diff
@@ -237,10 +225,13 @@ static int test_siphash(int idx)
     size_t i;
 
     if (expectedlen != SIPHASH_MIN_DIGEST_SIZE &&
-        expectedlen != SIPHASH_MAX_DIGEST_SIZE)
+        expectedlen != SIPHASH_MAX_DIGEST_SIZE) {
+        TEST_info("size %" OSSLzu " vs %d and %d", expectedlen,
+                  SIPHASH_MIN_DIGEST_SIZE, SIPHASH_MAX_DIGEST_SIZE);
```
Comment (levitte) [inline test/siphash_internal_test.c]: This is where C++ templates would come in handy...
(side note)

**Comment by levitte** [comment]
Merged.  a24c1e224338ccd3d2b831bd2e3ba52a1eb31f01

File: test/siphash_internal_test.c (hunk 1 above)
Comment (pauli-oracle) [inline test/siphash_internal_test.c]: Yeah, there are plenty of places where limited use of C++ features would be a boon.
The risk is overusing/overdoing the C++ features and making a huge unmaintainable mess.

And writing templates is convoluted.  It seems to turn into a game of who can create the best error message longer than a page :(

Comment (levitte) [inline test/siphash_internal_test.c]: > And writing templates is convoluted. It seems to turn into a game of who can create the best error message longer than a page :(

Hahaha, true dat!

**Pull request closed without merging**